src/exchange/exchange_client.rs

```diff
@@ -91,11 +93,34 @@ impl ExchangeClient {
             info.meta().await?
         };
 
+        let spot_meta = if let Some(spot_meta) = spot_meta {
+            spot_meta
+        } else {
+            let info = InfoClient::new(None, Some(base_url)).await?;
+            info.spot_meta().await?
+        };
+
         let mut coin_to_asset = HashMap::new();
         for (asset_ind, asset) in meta.universe.iter().enumerate() {
             coin_to_asset.insert(asset.name.clone(), asset_ind as u32);
         }
 
+        let parse_spot_name_to_u32 = |name: &str| {
+            name.replacen('@', "", 1).parse::<u32>().unwrap()
+        };
+
+        for (ind, asset) in spot_meta.universe.iter().enumerate() {
+            let mut spot_ind: u32 = ind as u32;
+
+            if let Some(0) = asset.name.find('@') {
+                spot_ind = 10000 + parse_spot_name_to_u32(&asset.name);
+            } else if asset.name == "PURR/USDC" {
+                spot_ind = 10000;
+            }
+
+            coin_to_asset.insert(asset.name.clone(), spot_ind);
```

Comment:
spot_index should just be 10000 plus the index returned in the spot meta

Reply:
Done